hibike/hibike_message.py

```diff
@@ -146,17 +329,60 @@ def make_error(error_code):
   message = HibikeMessage(messageTypes["Error"], payload)
   return message
 
-def make_ping():
-  """ Makes and returns Ping message."""
-  payload = bytearray()
-  message = HibikeMessage(messageTypes["Ping"], payload)
-  return message
+def parse_subscription_response(msg):
+  assert msg.getmessageID() == messageTypes["SubscriptionResponse"]
+  payload = msg.getPayload()
+  assert len(payload) == 15
+  params, delay, device_id, year, ID = struct.unpack("<HHHBQ", payload)
+  params = decode_params(device_id, params)
+  uid = (device_id << 72) | (year << 64) | ID
+  return (params, delay, uid)
+
+def parse_device_data(msg, device_id):
+  assert msg.getmessageID() == messageTypes["DeviceData"]
+  payload = msg.getPayload()
+  assert len(payload) >= 2
+  params,  =  struct.unpack("<H", payload[:2])
+  params = decode_params(device_id, params)
+  struct_format = "<"
+  for param in params:
+    struct_format += paramTypes[paramMap[device_id][param][1]]
```

Comment:
Construction of a type string from param types should be abstracted into a function.